src/core/error.js

```diff
@@ -81,3 +81,20 @@ export function rethrowAsync(var_args) {
     throw error;
   });
 }
+
+/**
+ * Executes the provided callback in a try/catch and rethrows any errors
+ * asynchronously.
+ *
+ * @param {function(...*):T} callback
+ * @param {...*} var_args
+ * @return {T}
+ * @template T
+ */
+export function tryCallback(callback, var_args) {
```

Comment:
Using a `...args` here is one of the few exceptions, because we explicitly need to remove the first argument. We either construct an array now, or do it at the call site, and it's prettier here.
```suggestion
export function tryCallback(callback, ...args) {
```